rpc/src/v1/types/transaction.rs

```diff
@@ -51,6 +51,9 @@ pub struct Transaction {
 	pub creates: Option<H160>,
 	/// Raw transaction data
 	pub raw: Bytes,
+	/// Public key of the signer.
+	#[serde(rename="publicKey")]
+	pub public_key: Option<H512>,
```

Comment:
it's additive functionality and therefore of no great harm. in any case, we're doing it already with additional APIs and providing a separate API for it would be substantially more code/complexity with relatively little gain.